File: INICIANTE/1117/main.dart
import 'dart:io';

double average() {
  double grade1 = double.parse(stdin.readLineSync()!);
  while (grade1 < 1 || grade1 > 10) {
    print('nota invalida');
    grade1 = double.parse(stdin.readLineSync()!);
  }
  double grade2 = double.parse(stdin.readLineSync()!);
  while (grade2 < 1 || grade2 > 10) {
    print('nota invalida');
    grade2 = double.parse(stdin.readLineSync()!);
  }
  return (grade1 + grade2) / 2;
}

void main() {
  print('media = ${average().toStringAsFixed(2)}');
}

File: INICIANTE/1117/main.py
def average():
    grade_1 = float(input())
    while grade_1 < 0 or grade_1 > 10:
        print('nota invalida')
        grade_1 = float(input())
    grade_2 = float(input())
    while grade_2 < 0 or grade_2 > 10:
        print('nota invalida')
        grade_2 = float(input())
    return (grade_1 + grade_2) / 2


print(f'media = {average():.2f}')
